base_user_role/models/role.py

```diff
@@ -49,6 +51,7 @@ def _compute_user_ids(self):
 
     @api.model
     def create(self, vals):
+        vals.update({"category_id": self._default_category_id().id})
```

Comment:
Please, only update `vals` if no value is present yet. In my own setup I create roles with a category already to make up for this bug.

Reply:
Maybe also check if the group has a category already.

Reply:
> only update vals if no value is present yet

done. Thanks.

> Maybe also check if the group has a category already.

Not sure to understand. group is created when role is created (via ``    _inherits = {"res.groups": "group_id"}``). 
Do you want I handle the case where create is called with a ``group_id`` that has a ``category_id`` defined ?

Reply:
On the contrary. In our module, we have
```
    <record id="group_role_something" model="res.groups">
        <field name="name">Role Someting</field>
        <field name="category_id" ref="module_category_roles"/>
        <field name="implied_ids">
        ....
        </field>
    </record>
    <record id="role_something" model="res.users.role">
        <field name="group_id" ref="module.group_role_something"/>
    </record>
```

Reply:
1) OK. I had not thought of such a design. Naive question, why don't you just define roles ? (without groups, that will be created automatically)
something like : 
```
    <record id="role_something" model="res.users.role">
        <field name="name">Role Someting</field>
        <field name="line_ids">
        ....
        </field>
    </record>
```

2) I updated my code. I don't affect "category_id" if "group_id" is defined. (because I suppose that the group is correctly set, and should not be updated). Is it OK for you ?

Reply:
Thank you!

We create the group separately because we like to have explicit group xml ids so that we can use them in access rules and view definitions.